presto-iceberg/pom.xml

```diff
@@ -511,23 +511,17 @@
             <artifactId>presto-cache</artifactId>
             <scope>compile</scope>
         </dependency>
-        
         <dependency>
             <groupId>com.facebook.presto</groupId>
             <artifactId>presto-main</artifactId>
-            <scope>test</scope>
         </dependency>
-
         <dependency>
-          <groupId>com.facebook.presto</groupId>
-          <artifactId>presto-parser</artifactId>
-          <scope>test</scope>
+            <groupId>com.facebook.presto</groupId>
+            <artifactId>presto-parser</artifactId>
         </dependency>
```

Comment:
I'm a little concern that we have to depend on `presto-main` in this change, as well as `presto-parser` and `presto-analyzer`. Have you considered some way, maybe like re-factoring or something else, to avoid depending on `presto-main`?

Reply:
I will see about refactoring

Reply:
I refactored parts to `presto-common` and `presto-spi`. It was a little tricky because previously I relied on some guava bits so I needed to re-implement some small utilities in these modules